src/test/java/com/juniorstart/juniorstart/controller/UserControllerTest.java

```diff
@@ -1,39 +1,187 @@
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.*;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.test.web.servlet.MockMvc;
-import org.springframework.test.web.servlet.MockMvcBuilder;
+import org.springframework.test.web.servlet.MvcResult;
 import org.springframework.test.web.servlet.setup.MockMvcBuilders;
 
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.Is.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/** Test for UserController class.
+ * @author Noboseki
+ * @version 1.0
+ * @since 1.0
+ */
 @ExtendWith(MockitoExtension.class)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 class UserControllerTest {
     @Mock
     UserService userService;
-
-    @Mock
-    ChangeMailRequest changeMailRequest;
-
     @InjectMocks
     UserController userController;
 
+    Gson gson = new Gson();
     MockMvc mockMvc;
 
+    ResponseEntity<ApiResponse> response = ResponseEntity.ok((new ApiResponse(true,"Test Ok")));
+    ChangePasswordRequest changePasswordRequest = new ChangePasswordRequest("Password 2",UUID.randomUUID(), "Password");
+    ChangeMailRequest changeMailRequest = new ChangeMailRequest("test@tes.com", UUID.randomUUID(), "Password");
+    ChangeStatusRequest changeStatusRequest = new ChangeStatusRequest(UserStatus.OPEN, UUID.randomUUID(), "Password");
+
+    String changePasswordJson = gson.toJson(changePasswordRequest);
+    String changeEmailJson = gson.toJson(changeMailRequest);
+    String changeStatusJson = gson.toJson(changeStatusRequest);
+
     @BeforeEach
     void setUp() {
         mockMvc = MockMvcBuilders.standaloneSetup(userController).build();
     }
 
     @Test
-    void name() throws Exception {
+    @Order(1)
+    @DisplayName("Change password Correct")
+    void testChangePasswordCorrect() throws Exception {
+        //When
+        when(userService.changePassword(any(ChangePasswordRequest.class))).thenReturn(response);
+
+        //Then
+        correctMvcPerform("/changePassword", changePasswordJson);
```

Comment:
Super, ale można można dodać testy z spring security :)

Reply:
A to nie byłyby już testy integracyjne, a nie jednostkowe ? To w późniejszym etapie można dodać.

Reply:
Sprawdziłem dokładnie, w  jednostkowym teście kontrolera nie można dodać springSecurity context. Musi to być test integracyjny, a to już jest inna bajka.

Reply:
A zadziało to jeżeli doda się zabezpieczenia dla metod?